cmd/boots/tftp_test.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"testing"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+func TestExtractTraceparentFromFilename(t *testing.T) {
+	tests := map[string]struct {
+		fileIn  string
+		fileOut string
+		err     error
+		spanId  string
+		traceId string
+	}{
+		"do nothing when no tp": {fileIn: "undionly.ipxe", fileOut: "undionly.ipxe", err: nil},
+		"ignore bad filename": {
+			fileIn:  "undionly.ipxe-00-0000-0000-00",
+			fileOut: "undionly.ipxe-00-0000-0000-00",
+			err:     nil,
+		},
+		"ignore corrupt tp": {
+			fileIn:  "undionly.ipxe-00-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx-abcdefghijklmnop-01",
```

Comment:
This is not for this PR but we should be careful about the length of the name of the file. I believe for `Boot file name` field is maxed out at 128 octets/chars. 

https://www.apc.com/us/en/faqs/FA156110/
https://www.ietf.org/rfc/rfc2131.txt - see page 8, 9

Reply:
Good catch, https://www.ietf.org/rfc/rfc2131.html#section-2.

Reply:
I worried about that but there wasn't another way to do it without caching trace state which is even more naughty.

Reply:
The traceparent is 56 chars, 57 with the leading dash added here. I don't see any filenames in our code being requested that are long enough to worry. This functionality is kinda dubious anyways so if we want to remove it I'm fine with that.